tests/MongoDB.Driver.Core.Tests/Core/Compression/CompressorsTests.cs

```diff
@@ -0,0 +1,170 @@
+                    }
+                    else
+                    {
+                        output.Length.Should().Be(input.Length + zlibHeaderSize);
+                    }
+                    input.Position = 0;
+                    input.SetLength(0);
+                    output.Position = 0;
+                    compressor.Decompress(output, input);
+                },
+                (input, output) =>
+                {
+                    input.Position = 0;
+                    var result = Encoding.ASCII.GetString(input.ReadBytes((int)input.Length));
+                    result.Should().Be(__testMessage);
+                });
+        }
+
+        [Theory]
+        [InlineData(CompressorType.Snappy)]
+        public void Compressor_should_read_the_previously_written_message_or_throw_the_exception_if_the_current_platform_is_not_supported(CompressorType compressorType)
+        {
+            var bytes = Encoding.ASCII.GetBytes(__testMessage);
+            var compressor = GetCompressor(compressorType);
+#if NET452 || NETSTANDARD2_0
+            Assert(
+                bytes,
+                (input, output) =>
+                {
+                    compressor.Compress(input, output);
+                    input.Length.Should().BeGreaterThan(output.Length);
+                    input.Position = 0;
+                    input.SetLength(0);
+                    output.Position = 0;
+                    compressor.Decompress(output, input);
+                },
+                (input, output) =>
+                {
+                    input.Position = 0;
+                    var result = Encoding.ASCII.GetString(input.ReadBytes((int)input.Length));
+                    result.Should().Be(__testMessage);
+                });
+#else
+            var exception = Record.Exception(() => { compressor.Compress(Mock.Of<Stream>(), Mock.Of<Stream>()); });
+            exception.Should().BeOfType<NotSupportedException>();
+            
+            exception = Record.Exception(() => { compressor.Decompress(Mock.Of<Stream>(), Mock.Of<Stream>()); });
+            exception.Should().BeOfType<NotSupportedException>();
+#endif
+        }
+
+        [Fact]
+        public void Zlib_should_generate_expected_compressed_bytes()
+        {
+            var bytes = Encoding.ASCII.GetBytes(__testMessage);
+            Assert(
+                bytes,
+                (input, output) =>
+                {
+                    var compressor = GetCompressor(CompressorType.Zlib, 6);
+                    compressor.Compress(input, output);
+                },
+                (input, output) =>
+                {
+                    var resultBytes = output.ToArray();
+                    var result = string.Join(",", resultBytes);
+                    result
+                        .Should()
+                        .Be("120,156,74,76,74,78,73,77,75,207,200,204,202,206,201,205,203,47,40,44,42,46,41,45,43,175,168,172,74,36,67,6,0,0,0,255,255,3,0,21,79,33,94");
```

Comment:
I've added this line to ensure that the zlib implementation returns the result which is allowed by the server.